test/cpp/api/rnn.cpp

```diff
@@ -247,3 +248,250 @@ TEST_F(RNNTest, BidirectionalFlattenParameters) {
   GRU gru(GRUOptions(100, 256).layers(2).bidirectional(true));
   gru->flatten_parameters();
 }
+
+// This test is a port of python code introduced here:
+// https://towardsdatascience.com/understanding-bidirectional-rnn-in-pytorch-5bd25a5dd66
+// Reverse forward of bidrectional GRU should act
+// as regular forward of unidirectional GRU
+void BidirectionalGRUReverseForward(bool cuda) {
+  auto options = torch::TensorOptions()
+                        .dtype(torch::kFloat32).requires_grad(false);
+  auto input =
+              torch::tensor({1, 2, 3, 4, 5}, options).reshape({5, 1, 1});
+  auto input_reversed =
+              torch::tensor({5, 4, 3, 2, 1}, options).reshape({5, 1, 1});
+
+  GRU bi_grus {GRUOptions(1, 1)
+                  .layers(1).batch_first(false).bidirectional(true)};
+  GRU reverse_gru {GRUOptions(1, 1)
+                  .layers(1).batch_first(false).bidirectional(false)};
+
+  if (cuda) {
+    bi_grus->to(torch::kCUDA);
+    reverse_gru->to(torch::kCUDA);
+    input = input.to(torch::kCUDA);
```

Comment:
It's more efficient to create what you can on the GPU instead of moving things. `input` and `input_reversed` can be created on the GPU:
```
  auto options = torch::TensorOptions()
                        .dtype(torch::kFloat32).requires_grad(false).device(cuda ? kCUDA : kCPU);
```